### ISSUE TITLE
Bundler issues incorrect warning about the bundle install --no-cache flag

### ISSUE THREAD

**Issue by eregon** [rubygems/rubygems#4050]
### Describe the problem as clearly as you can

Bundler 2.1.4 warns that the `bundle install` flag `--no-cache` flag is deprecated because "it relies on being remembered", but `--no-cache` is actually not a remembered flag (from what I can tell).

### Post steps to reproduce the problem

```ruby
source 'https://rubygems.org'

gem 'rspec', '~> 3.0'
```

```
$ bundle install --local --no-cache
[DEPRECATED] The `--no-cache` flag is deprecated because it relies on being remembered across bundler invocations, which bundler will no longer do in future versions. Instead please use `bundle config set no-cache 'true'`, and stop using this flag
...

$ cat .bundle/config
cat: .bundle/config: No such file or directory

Or if there is some other setting:
$ cat .bundle/config
---
BUNDLE_CACHE_PATH: "..."
```

Note there is no BUNDLE_NO_CACHE or BUNDLE_NO-CACHE entry.

Also, the recommended command:
```
bundle config set no-cache 'true'
```
is incorrect because:
* it's not local (but I fixed that on master a while ago)
* it would set a BUNDLE_NO-CACHE key and not BUNDLE_NO_CACHE
* it's not a remembered option anyway so probably this doesn't matter

## Environment

```
Bundler       2.1.4
  Platforms   ruby, x86_64-linux
Ruby          2.7.2p137 (2020-10-01 revision 5445e0435260b449decf2ac16f9d09bae3cafe72) [x86_64-linux]
  Full Path   /home/eregon/.rubies/ruby-2.7.2/bin/ruby
  Config Dir  /home/eregon/.rubies/ruby-2.7.2/etc
RubyGems      3.1.4
  Gem Home    /home/eregon/.rubies/ruby-2.7.2/lib/ruby/gems/2.7.0
  Gem Path    /home/eregon/.gem/ruby/2.7.0:/home/eregon/.rubies/ruby-2.7.2/lib/ruby/gems/2.7.0
  User Home   /home/eregon
  User Path   /home/eregon/.gem/ruby/2.7.0
  Bin Dir     /home/eregon/.rubies/ruby-2.7.2/bin
Tools         
  Git         2.26.2
  RVM         not installed
  rbenv       not installed
  chruby      0.3.9
```

## Bundler Build Metadata

```
Built At          2020-11-05
Git SHA           unknown
Released Version  false
```

## Bundler settings

```
cache_path
  Set for your local app (/home/eregon/code/truffleruby-ws/truffleruby/spec/ffi/.bundle/config): "/home/eregon/code/truffleruby-ws/truffleruby/truffleruby-gem-test-pack/gem-testing/msgpack-ruby/gem-cache"
```

### Gemfile

```ruby
source 'https://rubygems.org'

gem 'rspec', '~> 3.0'
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    diff-lcs (1.3)
    rspec (3.8.0)
      rspec-core (~> 3.8.0)
      rspec-expectations (~> 3.8.0)
      rspec-mocks (~> 3.8.0)
    rspec-core (3.8.0)
      rspec-support (~> 3.8.0)
    rspec-expectations (3.8.2)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.8.0)
    rspec-mocks (3.8.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.8.0)
    rspec-support (3.8.0)

PLATFORMS
  ruby

DEPENDENCIES
  rspec (~> 3.0)

BUNDLED WITH
   2.1.4
```

**Comment by deivid-rodriguez** [comment]
Already fixed in 2.2.0.rc.1 by #3623 and #3688.

**Comment by eregon** [comment]
Great, thanks!
